rust/signed_doc/tests/decoding.rs

```diff
@@ -20,6 +20,67 @@ struct TestCase {
     post_checks: Option<Box<PostCheck>>,
 }
 
+// `parameters` value along with its aliases are not allowed to be presented
+#[allow(clippy::unwrap_used)]
+fn signed_doc_with_parameters_and_aliases(aliases: Vec<String>, valid: bool) -> TestCase {
+    let uuid_v7 = UuidV7::new();
+    let uuid_v4 = UuidV4::new();
+    let doc_ref = DocumentRef::new(UuidV7::new(), UuidV7::new(), DocLocator::default());
+
+    TestCase {
+        name: "Multiple definitions of campaign_id, brand_id, category_id and parameters at once. [INVALID]",
+        bytes_gen: Box::new({
+            move || {
+                let (_, _, kid) = common::create_dummy_key_pair(RoleId::Role0).unwrap();
+
+                let mut e = Encoder::new(Vec::new());
+                e.tag(Tag::new(98))?;
+                e.array(4)?;
+
+                // protected headers (metadata fields)
+                e.bytes({
+                    let mut p_headers = Encoder::new(Vec::new());
+                    p_headers.map(5u64.checked_add(u64::try_from(aliases.len()).unwrap()).unwrap())?;
+                    p_headers.u8(3)?.encode(ContentType::Json)?;
+                    p_headers.str("type")?.encode_with(uuid_v4, &mut catalyst_types::uuid::CborContext::Tagged)?;
+                    p_headers.str("id")?.encode_with(uuid_v7, &mut catalyst_types::uuid::CborContext::Tagged)?;
+                    p_headers.str("ver")?.encode_with(uuid_v7, &mut catalyst_types::uuid::CborContext::Tagged)?;
+                    p_headers.str("parameters")?.encode_with(doc_ref.clone(), &mut ())?;
+
+                    for alias in aliases.clone() {
+                        p_headers.str(&alias)?.encode_with(doc_ref.clone(), &mut ())?;
+                    }
+
+                    p_headers.into_writer().as_slice()
+                })?;
+
+                // empty unprotected headers
+                e.map(0)?;
+                // content
+                e.bytes(serde_json::to_vec(&serde_json::Value::Null)?.as_slice())?;
+                // signatures
+                // one signature
```

Comment:
For the simplicity of this test, I think its not necessary to encode a signatures, lets leave a signature array empty.